cli/waiter/subcommands/create.py

```diff
@@ -0,0 +1,131 @@
+            raise Exception('You must either specify a cluster via --cluster or set "default-for-create" to true for '
+                            'one of your configured clusters.')
+        elif num_default_create_clusters > 1:
+            raise Exception('You have "default-for-create" set to true for more than one cluster.')
+        else:
+            cluster = default_for_create[0]
+    else:
+        cluster = clusters[0]
+    token_name = args.pop('token')
+    token_fields = args
+    return create_or_update(cluster, token_name, token_fields)
+
+
+def register(add_parser):
+    """Adds this sub-command's parser and returns the action function"""
+    global create_parser
+    create_parser = add_parser('create', help='create token', description='Create or update a Waiter token. '
+                                                                          'In addition to the optional arguments '
+                                                                          'explicitly listed below, '
+                                                                          'you can optionally provide any Waiter '
+                                                                          'token parameter as a flag. For example, '
+                                                                          'to specify 10 seconds for the '
+                                                                          'grace-period-secs parameter, '
+                                                                          'you can pass --grace-period-secs 10.')
+    create_parser.add_argument('--name', '-n', help='name of service')
+    create_parser.add_argument('--version', '-v', help='version of service')
+    create_parser.add_argument('--cmd', '-C', help='command to start service')
+    create_parser.add_argument('--cmd-type', '-t', help='command type of service (e.g. "shell")', dest='cmd-type')
+    create_parser.add_argument('--cpus', '-c', help='cpus to reserve for service', type=float)
+    create_parser.add_argument('--mem', '-m', help='memory to reserve for service', type=int)
+    create_parser.add_argument('--ports', help='number of ports to reserve for service', type=int)
+    create_parser.add_argument('token', nargs=1)
+    return create
+
+
+def str2bool(v):
+    """Converts the given string to a boolean, or raises"""
+    if v.lower() in TRUE_STRINGS:
+        return True
+    elif v.lower() in FALSE_STRINGS:
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+
+def add_implicit_arguments(unknown_args):
+    """
+    Given the list of "unknown" args, dynamically adds proper arguments to
+    the subparser, allowing us to support any token parameter as a flag
+    """
+    num_unknown_args = len(unknown_args)
+    for i in range(num_unknown_args):
+        arg = unknown_args[i]
+        if arg.startswith(("-", "--")):
+            if arg.endswith('-secs'):
```

Comment:
- Some others also belong in this category: `-mins`, `-instances`. 
- `-factor` needs to be double.

Reply:
Done